src/serde.rs

```diff
@@ -0,0 +1,336 @@
+//! Serde support for the `Verbosity` type.
+//!
+//! This module provides serialization and deserialization support for the `Verbosity` type.
+//!
+//! The verbosity is serialized as an optional log level that represents the equivalent log level
+//! given the number of verbose and quiet flags. When deserializing, the log level is converted into
+//! a number of verbose and quiet flags, which keeps one of the flags at 0 and the other at the
+//! difference between the log level and the default log level.
+use std::{fmt, marker::PhantomData};
+
+use log::Level;
+use serde::{de, Deserialize, Deserializer, Serialize};
+
+use crate::{level_value, LogLevel, Verbosity};
+
+#[cfg(feature = "serde")]
+impl<L: LogLevel> Serialize for Verbosity<L> {
+    fn serialize<S: serde::Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        match self.log_level() {
+            None => serializer.serialize_none(),
+            Some(ref level) => serializer.serialize_some(level),
+        }
+    }
+}
+
+impl<'de, L: LogLevel> Deserialize<'de> for Verbosity<L> {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        struct VerbosityVisitor<L>(PhantomData<L>);
+        impl<'de, L: LogLevel> de::Visitor<'de> for VerbosityVisitor<L> {
+            type Value = Verbosity<L>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+                formatter.write_str("a log level")
+            }
+
+            fn visit_none<E: de::Error>(self) -> Result<Self::Value, E> {
+                let verbose = 0;
+                let quiet = (level_value(L::default()) + 1) as u8;
+                Ok(Verbosity::new(verbose, quiet))
+            }
+
+            fn visit_some<D: Deserializer<'de>>(
+                self,
+                deserializer: D,
+            ) -> Result<Self::Value, D::Error> {
+                let level = Level::deserialize(deserializer)?;
+                let value = level_value(Some(level)) - level_value(L::default());
+                let verbose = value.max(0) as u8;
+                let quiet = (-value).max(0) as u8;
+                Ok(Verbosity::new(verbose, quiet))
+            }
+        }
```

Comment:
Please split this off into its own private construcor for taking an `Option<Level>` with tests for that

Reply:
No longer necessary in 2dda2ab